gobblin-kafka/src/main/java/gobblin/kafka/writer/KafkaDataWriter.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.kafka.writer;
+
+import java.io.IOException;
+import java.util.Properties;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.apache.kafka.clients.producer.Callback;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.clients.producer.RecordMetadata;
+
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
+
+import lombok.extern.slf4j.Slf4j;
+
+import gobblin.writer.DataWriter;
+
+
+/**
+ * Implementation of KafkaWriter that wraps a {@link KafkaProducer}.
+ * This does not provide transactional / exactly-once semantics.
+ * Applications should expect data to be possibly written to Kafka even if the overall Gobblin job fails.
+ *
+ */
+@Slf4j
+public class KafkaDataWriter<D> implements DataWriter<D> {
+
+  private final KafkaProducer<String, D> producer;
+  private final String topic;
+  private final AtomicInteger recordsWritten;
+  private final AtomicInteger bytesWritten;
+
+
+  private static final String KEY_SERIALIZER_CONFIG = "key.serializer";
+  private static final String DEFAULT_KEY_SERIALIZER = "org.apache.kafka.common.serialization.StringSerializer";
+  private static final String VALUE_SERIALIZER_CONFIG = "value.serializer";
+  private static final String DEFAULT_VALUE_SERIALIZER = "org.apache.kafka.common.serialization.ByteArraySerializer";
+
+  public KafkaDataWriter(Properties props) {
+    recordsWritten = new AtomicInteger(0);
+    bytesWritten = new AtomicInteger(0);
```

Comment:
Is there a way to accurately track this when writing records? Right now it doesn't seem to be updated.